zephyr/test/vmh.c

```diff
@@ -17,8 +18,258 @@
 
 LOG_MODULE_DECLARE(sof_boot_test, CONFIG_SOF_LOG_LEVEL);
 
+/* Test creating and freeing a virtual memory heap */
+static void test_vmh_init_and_free_heap(int memory_region_attribute,
+					struct vmh_heap_config *config,
+					int core_id,
+					bool allocating_continuously,
+					bool expect_success)
+{
+	struct vmh_heap *heap = vmh_init_heap(config, memory_region_attribute,
+		core_id, allocating_continuously);
+	if (expect_success) {
+		zassert_not_null(heap,
+		"Heap initialization expected to succeed but failed");
+		}
+	else
+		zassert_is_null(heap, "Heap initialization expected to fail but succeeded");
+
+	if (heap) {
+		int ret = vmh_free_heap(heap);
+
+		zassert_equal(ret, 0, "Failed to free heap");
+	}
+}
```

Comment:
can we have all this as a change to the current version instead of a rewrite?  It would be much easier to review changes than complete new code.

Reply:
Old code was not reused in any way in the vmh.c file so the change would only confuse the reader who did not write this code before. The new approach enhances existing tests but does it in a parametrized fashion and I see no way to do it as a change in a readable way.
I specifically did it as a rewrite to make it clear in the review.